libpod/image/image.go

```diff
@@ -469,7 +469,7 @@ func (ir *Runtime) getLocalImage(inputName string) (string, *storage.Image, erro
 		if err != nil {
 			return "", nil, err
 		}
-		img, err := ir.store.Image(ref.String())
+		img, err := ir.store.Image(reference.TagNameOnly(ref).String())
```

Comment:
@baude @mheon ... this should have a quite positive effect on container creation. If the user input wasn't tagged or digested, this lookup always failed, so we fell back to looking at all tags of all images (below in the code).

Now, we normalize to ":latest" which should have a noticeable performance boost on nodes with many images. It's also fixed in the short-names PR but it was necessary to get this fix working as well.